src/sys/socket/addr.rs

```diff
@@ -15,12 +17,24 @@ use std::os::unix::ffi::OsStrExt;
  *
  */
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+#[repr(C)]
+pub struct sockaddr_nl {
```

Comment:
I think it still makes sense to use the struct from libc and implement the required traits as in:

```diff
diff --git a/src/sys/socket/addr.rs b/src/sys/socket/addr.rs
index eadd990..3728eb2 100644
--- a/src/sys/socket/addr.rs
+++ b/src/sys/socket/addr.rs
@@ -562,10 +562,29 @@ pub mod netlink {
     use ::sys::socket::addr::{AddressFamily,sockaddr_nl};
     use libc::sa_family_t;
     use std::fmt;
+    use std::hash::{Hash, Hasher};
 
-    #[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+    #[derive(Copy, Clone)]
     pub struct NetlinkAddr(pub sockaddr_nl);
 
+    //, PartialEq, Eq, Debug, Hash
+    impl PartialEq for NetlinkAddr {
+        fn eq(&self, other :&Self) -> bool {
+            let (inner, other) = (self.0, other.0);
+            (inner.nl_family, inner.nl_pid, inner.nl_groups) == (other.nl_family, other.nl_pid, other.nl_groups)
+        }
+    }
+
+    impl Eq for NetlinkAddr {}
+
+    impl Hash for NetlinkAddr {
+        fn hash<H: Hasher>(&self, s: &mut H) {
+            let inner = self.0;
+            (inner.nl_family, inner.nl_pid, inner.nl_groups).hash(s);
+        }
+    }
+
+
     impl NetlinkAddr {
         pub fn new(pid: u32, groups: u32) -> NetlinkAddr {
             NetlinkAddr(sockaddr_nl {
```